src/watertap_contrib/seto/unit_models/zero_order/chemical_softening_0D.py

```diff
@@ -0,0 +1,949 @@
+from copy import deepcopy
+
+# check
+# Import Pyomo libraries
+from pyomo.environ import (
+    Set,
+    Var,
+    Constraint,
+    check_optimal_termination,
+    Param,
+    Suffix,
+    value,
+    log,
+    units as pyunits,
+    Expression,
+    NonNegativeReals,
+)
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    UnitModelBlockData,
+    useDefault,
+)
+from idaes.core.solvers.get_solver import get_solver
+from idaes.core.util.tables import create_stream_table_dataframe
+from idaes.core.util.constants import Constants
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.misc import StrEnum
+
+from idaes.core.util.exceptions import ConfigurationError, InitializationError
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+
+from watertap.core import InitializationMixin
+
+__author__ = "Kurban Sitterley, Abdiel Lugo"
+
+_log = idaeslog.getLogger(__name__)
+
+
+class SofteningTrainType(StrEnum):
```

Comment:
I don't see this CONFIG argument being used anywhere in the unit model. If it isn't necessary, you should remove it.

Reply:
I was about to agree but then came to the realization that this is still a draft PR, so perhaps @MuktaHardikar has some plans for this StrEnum class

Reply:
I had this here incase we wanted to add the other softening train type. But I am removing it for now.